### PR TITLE
openfortivpn: `configure --enable-legacy-pppd`

### LINKED ISSUE #154319: Please adapt openfortivpn 1.21.0 formula: add `--enable-legacy-pppd`
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
```

### Verification

- [ ] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [ ] I ran `brew update` and am still able to reproduce my issue.
- [ ] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [ ] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

Reporting a packaging issue in the openfortivpn formula. I am maintaining openfortivpn.

### What happened (include all command output)?

See https://github.com/adrienverge/openfortivpn/issues/1165.

### What did you expect to happen?

See https://github.com/adrienverge/openfortivpn/issues/1165.

### Step-by-step reproduction instructions (by running `brew` commands)

See https://github.com/adrienverge/openfortivpn/issues/1165.

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Add the system `pppd` executable as a dependency.

The configure option `--enable-legacy-pppd` is required for pppd < 2.5.0. As soon as a new macOS version ships pppd ≥ 2.5.0, you will have to add code to:
- remove `--enable-legacy-pppd` for new targets that ship pppd ≥ 2.5.0;
- keep `--enable-legacy-pppd` for old targets that still ship pppd < 2.5.0.

See:
https://github.com/adrienverge/openfortivpn/pull/1148
https://github.com/adrienverge/openfortivpn/pull/1159

Fixes #154319.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/o/openfortivpn.rb [hunk 1]
```diff
@@ -15,6 +15,8 @@ class Openfortivpn < Formula
     sha256 x86_64_linux:   "a9fddfe0c49d465a13e5492612ccc2ced64b88e428dc3ab955218a11eee6ba3c"
   end
 
+  uses_from_macos "pppd"
```
Comment (DimitriPapadopoulos) [inline Formula/o/openfortivpn.rb:18]: Is that the proper way to add a dependency on the system `pppd` executable?

File: Formula/o/openfortivpn.rb [hunk 2]
```diff
@@ -20,10 +20,13 @@ class Openfortivpn < Formula
   depends_on "pkg-config" => :build
   depends_on "openssl@3"
 
+  uses_from_macos "pppd"
```
Comment (SMillerDev) [inline Formula/o/openfortivpn.rb:23]: There is no formula for that yet, so maybe just comment it out?
```suggestion
  # awaiting formula creation
  # uses_from_macos "pppd"
```

Comment (DimitriPapadopoulos) [reply Formula/o/openfortivpn.rb:23]: Sure, I can do that. Time permitting, I could try to create a “system” formula for ppp/pppd, based on other system formulae such as [Formula/z/zlib.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/z/zlib.rb), but that's probably beyond what I can do without owning a mac.

**Review by SMillerDev — APPROVED** [review]

**Comment by SMillerDev** [comment]
Thanks @DimitriPapadopoulos  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by Neustradamus** [comment]
To follow this PR.

**Pull request merged**